clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -1314,13 +1332,36 @@ private void recordTopicFetchMetrics(String topic, int bytes, int records) {
         private void updatePartitionLagSensors(Set<TopicPartition> assignedPartitions) {
             if (this.assignedPartitions != null) {
                 for (TopicPartition tp : this.assignedPartitions) {
-                    if (!assignedPartitions.contains(tp))
+                    if (!assignedPartitions.contains(tp)) {
                         metrics.removeSensor(partitionLagMetricName(tp));
+                    }
                 }
             }
             this.assignedPartitions = assignedPartitions;
         }
 
+        private void updatePartitionLeadSensors() {
+            metrics.removeSensor(partitionLeadSensorName());
+            this.leadMetricAddedPartitions.clear();
```

Comment:
It seems that it's better to be consistent with updatePartitionLagSensors() by only removing partitions no longer assigned?